soccerdata/sofascore.py

```diff
@@ -0,0 +1,265 @@
+        Seasons to include. Supports multiple formats.
+        Examples: '16-17'; 2016; '2016-17'; [14, 15, 16]
+    proxy : 'tor' or dict or list(dict) or callable, optional
+        Use a proxy to hide your IP address. Valid options are:
+            - 'tor': Uses the Tor network. Tor should be running in
+              the background on port 9050.
+            - dict: A dictionary with the proxy to use. The dict should be
+              a mapping of supported protocols to proxy addresses. For example::
+
+                  {
+                      'http': 'http://10.10.1.10:3128',
+                      'https': 'http://10.10.1.10:1080',
+                  }
+
+            - list(dict): A list of proxies to choose from. A different proxy will
+              be selected from this list after failed requests, allowing rotating
+              proxies.
+            - callable: A function that returns a valid proxy. This function will
+              be called after failed requests, allowing rotating proxies.
+    no_cache : bool
+        If True, will not use cached data.
+    no_store : bool
+        If True, will not store downloaded data.
+    data_dir : Path
+        Path to directory where data will be cached.
+    """
+
+    def __init__(
+        self,
+        leagues: Optional[Union[str, List[str]]] = None,
+        seasons: Optional[Union[str, int, Iterable[Union[str, int]]]] = None,
+        proxy: Optional[
+            Union[str, Dict[str, str], List[Dict[str, str]], Callable[[], Dict[str, str]]]
+        ] = None,
+        no_cache: bool = NOCACHE,
+        no_store: bool = NOSTORE,
+        data_dir: Path = SOFASCORE_DATADIR,
+    ):
+        """Initialize the Sofascore reader."""
+        super().__init__(
+            leagues=leagues,
+            proxy=proxy,
+            no_cache=no_cache,
+            no_store=no_store,
+            data_dir=data_dir,
+        )
+        self.seasons = seasons  # type: ignore
+        if not self.no_store:
+            (self.data_dir / "leagues").mkdir(parents=True, exist_ok=True)
+            (self.data_dir / "seasons").mkdir(parents=True, exist_ok=True)
+            (self.data_dir / "matches").mkdir(parents=True, exist_ok=True)
+
+    def read_leagues(self) -> pd.DataFrame:
+        """Retrieve the selected leagues from the datasource.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        url = SOFASCORE_API + 'config/top-unique-tournaments/EN/football'
+        filepath = self.data_dir / 'leagues.json'
+        reader = self.get(url, filepath)
+        data = json.load(reader)
+        leagues = []
+        for k in data['uniqueTournaments']:
+            if k['name'] in BIG_FIVE_DICT:
+                leagues.append(
+                    {
+                        'region': BIG_FIVE_DICT[k['name']].split("-")[0],
+                        'league_id': k['id'],
+                        'league': BIG_FIVE_DICT[k['name']],
+                    }
+                )
```

Comment:
You should use the `_translate_league` helper instead.

Reply:
Included now